plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/LibraryClasspathContainerSerializer.java

```diff
@@ -0,0 +1,109 @@
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IJavaProject;
+import org.osgi.framework.FrameworkUtil;
+
+public class LibraryClasspathContainerSerializer {
+
+  private static final Logger logger = Logger.getLogger(LibraryClasspathContainerSerializer.class.getName());
+
+  public interface LibraryContainerStateLocationProvider {
+    IPath getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) throws CoreException;
+  }
+
+  private LibraryContainerStateLocationProvider stateLocationProvider;
+  
+  public LibraryClasspathContainerSerializer() {
+    this(new DefaultStateLocationProvider());
+  }
+  
+  @VisibleForTesting
+  public LibraryClasspathContainerSerializer(LibraryContainerStateLocationProvider stateLocationProvider) {
+    this.stateLocationProvider = stateLocationProvider;
+  }
+
+  public void saveContainer(IJavaProject javaProject, LibraryClasspathContainer container) throws IOException,
+                                                                                                  CoreException {
+    File stateFile = getContainerStateFile(javaProject, container.getPath(), true);
+    if (stateFile == null) {
+      logger.warning("Container state file cannot be created, save failed");
+      return;
+    }
+    try (PrintWriter outputStream = new PrintWriter(stateFile)) {
+      Gson gson = new GsonBuilder().setPrettyPrinting().create();
+      outputStream.print(gson.toJson(new SerializableLibraryClasspathContainer(container)));
+    }
+  }
+
+  public LibraryClasspathContainer loadContainer(IJavaProject javaProject, IPath containerPath) throws IOException,
+                                                                                                       CoreException {
+    File stateFile = getContainerStateFile(javaProject, containerPath, false);
+    if (stateFile == null || !stateFile.exists()) {
+      return null;
+    }
+    try (FileReader fileReader = new FileReader(stateFile)) {
+      Gson gson = new GsonBuilder().create();
+      SerializableLibraryClasspathContainer fromJson =
+          gson.fromJson(fileReader, SerializableLibraryClasspathContainer.class);
+      return fromJson.toLibraryClasspathContainer();
+    }
+  }
+  
+  private File getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) 
+                                                                                                throws CoreException {
+    IPath containerStateFile = stateLocationProvider.getContainerStateFile(javaProject, containerPath, create);
+    if (containerStateFile != null) {
+      return containerStateFile.toFile();
+    } else {
+      return null;
+    }
```

Comment:
You preference, but what about `return containerStateFile == null ? null : containerStateFile.toFile();`?